src/components/PatientRecord/PatientRecord.tsx

```diff
@@ -206,4 +209,29 @@ const PathwayVisualizer: FC = () => {
   );
 };
 
+const McodeVisualizer: FC = () => {
+  const { mcodeRecords } = usePatientRecords();
+  type mcodeKey = keyof McodeElements & string;
+  const keysArray: mcodeKey[] = [];
+  for (const key in mcodeRecords) {
+    keysArray.push(key as mcodeKey);
+  }
+  return (
+    <table>
+      <tbody>
+        {keysArray.map(key => {
+          const result = key.replace(/([A-Z]+)*([A-Z][a-z])/g, '$1 $2');
```

Comment:
I'm not a regex expert and regexes can be hard to read so a comment would be helpful here.

Is there any reason we can't use a simpler regex like `/([A-Z])/g`?  the current one looks like it has an undefined first match baed on some quick testing.

Also this might be hard, but right now HER2 is displayed as `Her2 Receptor` and I think it should be `HER2 Receptor`.  Optimally we would have display values for these mcode elements set.

Reply:
Not entirely sure if this is legal from a TypeScript or style standpoint, but why not just make the field key directly match what we want the display value to be? Otherwise we need a brittle regex or a manual mapping of keys to field names.

i.e., in mcode.d.ts:
```ts
  export interface McodeElements {
    'Primary Cancer'?: string;
    'Laterality'?: string;
    'Tumor Category'?: string;
    'Node Category'?: string;
```

Reply:
Just did a quick test and this works. I think its the best solution because as you mentioned it doesn't require regex nor manual mapping

Reply:
I like this solution too

Reply:
I updated this on my branch and then made this branch use the keys